File: CS/T179722/AppointmentExchangeData.cs
using System;
using System.Linq;
using System.Runtime.InteropServices;
using System.Runtime.Serialization;
using System.Security;

namespace T179722 {
    [Serializable]
    [ComVisible(true)]
    [SecurityCritical]
    public class AppointmentExchangeData : ISerializable {
        public AppointmentExchangeData() {

        }

        public AppointmentExchangeData(SerializationInfo info, StreamingContext context) {
            Subject = (string)info.GetValue(nameof(Subject), typeof(string));
            Description = (string)info.GetValue(nameof(Description), typeof(string));
            LabelKey = (int)info.GetValue(nameof(LabelKey), typeof(int));
            StatusKey = (int)info.GetValue(nameof(StatusKey), typeof(int));
            Start = (DateTime)info.GetValue(nameof(Start), typeof(DateTime));
            Duration = (TimeSpan)info.GetValue(nameof(Duration), typeof(TimeSpan));

        }

        public string Subject { get; set; }
        public string Description { get; set; }
        public int LabelKey { get; set; }
        public int StatusKey { get; set; }
        public DateTime Start { get; set; }
        public TimeSpan Duration { get; set; }

        [SecurityCritical]
        public void GetObjectData(SerializationInfo info, StreamingContext context) {
            info.AddValue(nameof(Subject), Subject, typeof(string));
            info.AddValue(nameof(Description), Description, typeof(string));
            info.AddValue(nameof(LabelKey), LabelKey, typeof(int));
            info.AddValue(nameof(StatusKey), StatusKey, typeof(int));
            info.AddValue(nameof(Start), Start, typeof(DateTime));
            info.AddValue(nameof(Duration), Duration, typeof(TimeSpan));
        }

    }
}

File: VB/T179722/AppointmentExchangeData.vb
Imports System
Imports System.Runtime.InteropServices
Imports System.Runtime.Serialization
Imports System.Security

Namespace T179722

    <Serializable>
    <ComVisible(True)>
    <SecurityCritical>
    Public Class AppointmentExchangeData
        Implements ISerializable

        Public Sub New()
        End Sub

        Public Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
            Subject = CStr(info.GetValue(NameOf(AppointmentExchangeData.Subject), GetType(String)))
            Description = CStr(info.GetValue(NameOf(AppointmentExchangeData.Description), GetType(String)))
            LabelKey = CInt(info.GetValue(NameOf(AppointmentExchangeData.LabelKey), GetType(Integer)))
            StatusKey = CInt(info.GetValue(NameOf(AppointmentExchangeData.StatusKey), GetType(Integer)))
            Start = CDate(info.GetValue(NameOf(AppointmentExchangeData.Start), GetType(Date)))
            Duration = CType(info.GetValue(NameOf(AppointmentExchangeData.Duration), GetType(TimeSpan)), TimeSpan)
        End Sub

        Public Property Subject As String

        Public Property Description As String

        Public Property LabelKey As Integer

        Public Property StatusKey As Integer

        Public Property Start As Date

        Public Property Duration As TimeSpan

        <SecurityCritical>
        Public Sub GetObjectData(ByVal info As SerializationInfo, ByVal context As StreamingContext) Implements ISerializable.GetObjectData
            info.AddValue(NameOf(AppointmentExchangeData.Subject), Subject, GetType(String))
            info.AddValue(NameOf(AppointmentExchangeData.Description), Description, GetType(String))
            info.AddValue(NameOf(AppointmentExchangeData.LabelKey), LabelKey, GetType(Integer))
            info.AddValue(NameOf(AppointmentExchangeData.StatusKey), StatusKey, GetType(Integer))
            info.AddValue(NameOf(AppointmentExchangeData.Start), Start, GetType(Date))
            info.AddValue(NameOf(AppointmentExchangeData.Duration), Duration, GetType(TimeSpan))
        End Sub
    End Class
End Namespace
